File: altera/lib/core/constants/app_constants.dart
class AppConstants {
  // === FINANCE CATEGORIES ===
  static const List<String> financeCategories = [
    "Food",
    "Transportation",
    "Entertainment",
    "Shopping",
    "Health",
    "Education",
    "Utilities",
    "Gifts",
    "Invoice",
    "Others"
  ];

  // === TASK CATEGORIES ===
  static const List<String> taskCategories = [
    "Work",
    "Personal",
    "Study",
    "Shopping",
    "Health",
    "Education",
    "Others"
  ];

  // === GAMIFICATION ===
  static const int maxXpPerLevel = 100;
  static const int xpBonusBase = 5;
  static const int xpMultiplierPerDifficulty = 10;

  // === FIRESTORE PATHS ===
  static const String usersCollection = "users";
  static const String tasksSubcollection = "tasks";
  static const String transactionsSubcollection = "transactions";

  // === UI ===
  static const Duration splashDelay = Duration(milliseconds: 2000);

  // === ML NORMALIZATION ===
  static const double maxExpense = 64997.0;
  static const double maxBalance = 14213582.0;
  static const double maxTaskCount = 20.0;
  static const double maxDuration = 8.0;
  static const double maxSleep = 24.0;
  static const double maxDailyExpense = 249647.0;

  // === LOCAL STORAGE KEYS ===
  static const String prefName = "AlteraPrefs";
  static const String keyIsDarkTheme = "is_dark_theme";
  static const String keyUserToken = "user_token";
  static const String keyUserName = "user_name";
  static const String keyIsLoggedIn = "is_logged_in";

  // === APP INFO ===
  static const String appName = "ALTERA";
  static const String minPasswordLength = "6";
}

File: exorcist/app/src/main/java/com/example/coreyield/AppConstants.kt
package com.example.coreyield

object AppConstants {

    // === KEUANGAN (WAJIB ADA — dipakai di FinanceScreen.kt) ===
    val FINANCE_CATEGORIES = listOf(
        "Food",
        "Transportation",
        "Entertainment",
        "Shopping",
        "Health",
        "Education",
        "Utilities",
        "Gifts",
        "Invoice",
        "Others"
    )

    // === TUGAS (Opsional — untuk konsistensi) ===
    val TASK_CATEGORIES = listOf(
        "Work",
        "Personal",
        "Study",
        "Shopping",
        "Health",
        "Education",
        "Others"
    )

    // === GAMIFIKASI ===
    const val MAX_XP_PER_LEVEL = 100
    const val XP_BONUS_BASE = 5
    const val XP_MULTIPLIER_PER_DIFFICULTY = 10

    // === FIRESTORE PATHS ===
    const val USERS_COLLECTION = "users"
    const val TASKS_SUBCOLLECTION = "tasks"
    const val TRANSACTIONS_SUBCOLLECTION = "transactions"

    // === UI ===
    const val SPLASH_DELAY_MS = 2000L

    // === ML NORMALIZATION ===
    const val MAX_EXPENSE = 64997.0f
    const val MAX_BALANCE = 14213582.0f
    const val MAX_TASK_COUNT = 20.0f
    const val MAX_DURATION = 8.0f
    const val MAX_SLEEP = 24.0f
    const val MAX_DAILY_EXPENSE = 249647.0f

    // Nama File Penyimpanan Lokal (SharedPreferences)
    const val PREF_NAME = "CoreYieldPrefs"

    // Keys untuk Data Pengguna (Agar tersimpan)
    const val KEY_IS_DARK_THEME = "is_dark_theme"
    const val KEY_USER_TOKEN = "user_token"
    const val KEY_USER_NAME = "user_name"
    const val KEY_IS_LOGGED_IN = "is_logged_in"

    // Konstanta Lain (Opsional, sesuaikan dengan kebutuhan Anda)
    const val MIN_PASSWORD_LENGTH = 6
    const val BASE_URL = "https://api.coreyield.example.com/" // Jika ada API lain


}